lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -266,10 +264,11 @@ impl<'s> SegmentHolder {
         let mut applied_points = 0;
         for (idx, segment) in &self.segments {
             // Collect affected points first, we want to lock segment for writing as rare as possible
-            let segment_points = self.segment_points(ids, segment);
+            let segment_arc = segment.get();
```

Comment:
Looks like not-task-related code. But its ok, I see additional thread safety

Reply:
it was another possible reason for the same issue